server/server.ts

```diff
@@ -0,0 +1,126 @@
+import { Server } from "socket.io";
+import jwt from "jsonwebtoken";
+import mongoose from "mongoose";
+import http from "http";
+import app from "./index";
+import PrivateMessage from "./models/messageModel";
+import User from "./models/userModel";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+const MONGODB_URI = process.env.MONGODB_URI!;
+const AUTH_SECRET = process.env.AUTH_SECRET!;
+const PORT = process.env.APP_PORT || 4000;
+
+let cached = (global as any).mongoose;
```

Comment:
I'd just call it `mongooseInstance` or `cachedMongoose` - `cached` is harder to understand when someone just skims through the file.